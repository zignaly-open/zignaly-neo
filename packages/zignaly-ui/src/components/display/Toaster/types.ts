export interface ToasterProps {
  variant?: "success" | "error" | "info";
  size?: "large" | "small";
  caption?: string | JSX.Element;
  id?: string;
}
