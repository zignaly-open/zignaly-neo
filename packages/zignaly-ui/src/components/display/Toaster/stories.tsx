import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";
import Toaster from "./";

export default {
  title: "Display/Toaster",
  component: Toaster,
  argTypes: {
    variant: {
      options: ["success", "error"],
      control: { type: "select" },
    },
    size: {
      options: ["large", "small"],
      control: { type: "select" },
    },
    caption: {
      control: "text",
    },
  },
  parameters: {
    jest: ["test.tsx"],
  },
} as ComponentMeta<typeof Toaster>;

const Template: ComponentStory<typeof Toaster> = (args) => <Toaster {...args} />;

export const ExampleToaster = Template.bind({});
ExampleToaster.args = {
  caption: "Something went wrong",
};
