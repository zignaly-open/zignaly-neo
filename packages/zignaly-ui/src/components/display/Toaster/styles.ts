import styled, { css } from "styled-components";

export const Caption = styled.div`
  margin: 0 8px;
`;

export const IconContainer = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;

  svg {
    width: 100%;
    height: 100%;
  }
`;

export const ToastContainer = styled.div<{ variant: string; size: string }>`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  box-sizing: border-box;
  box-shadow: 4px 6px 4px ${(props) => props.theme.neutral900}44;
  border-radius: 5px;
  position: absolute;

  ${(props) =>
    props.size === "large" &&
    css`
      ${IconContainer} {
        width: 26px;
        height: 26px;
      }

      padding: 11px 24px;
      height: 48px;
    `}

  ${(props) =>
    props.size === "small" &&
    css`
      ${IconContainer} {
        width: 24px;
        height: 24px;
      }

      padding: 4px 18px;
      height: 32px;
    `}

  ${({ variant, theme: { palette, backgrounds } }) =>
    variant === "success" &&
    css`
      ${Caption} {
        color: ${palette.greenGraph};
      }

      border: 1px solid ${backgrounds.toastSuccess};
      background: ${backgrounds.toastSuccess};
    `}


  ${({ variant, theme: { palette } }) =>
    variant === "info" &&
    css`
      ${Caption} {
        color: ${palette.neutral200};
      }

      border: 1px solid ${palette.neutral600};
      background: ${palette.neutral600};
    `}

  ${({ variant, theme: { palette, backgrounds } }) =>
    variant === "error" &&
    css`
      ${Caption} {
        color: ${palette.redGraphOrError};
      }

      border: 1px solid ${backgrounds.toastError};
      background: ${backgrounds.toastError};
    `}
`;
