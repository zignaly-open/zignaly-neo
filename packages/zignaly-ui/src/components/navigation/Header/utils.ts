// TODO: Move this for config or theme folder.
export const headerConfig = {
  height: "52px",
};
