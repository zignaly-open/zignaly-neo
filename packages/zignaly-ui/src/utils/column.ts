import styled from "styled-components";

export const Column = styled.div<{
  gap?: number;
  justifyContent?: "center" | "end" | "start" | "space-between";
  alignItems?: "center" | "end" | "start" | "space-between";
  textAlign?: "end" | "center" | "start";
  height?: string;
  flex?: number;
  width?: string;
}>`
  display: flex;
  flex-direction: column;
  flex: ${(props) => props.flex};
  gap: ${(props) => props.gap}px;
  justify-content: ${(props) => props.justifyContent};
  align-items: ${(props) => props.alignItems};
  text-align: ${(props) => `${props.textAlign}`};
  height: ${(props) => `${props.height}`};
  width: ${(props) => `${props.width}`};
`;
