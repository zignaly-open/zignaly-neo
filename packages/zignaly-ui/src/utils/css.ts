import { css } from "styled-components";

export const NiceScrollbar = css`
  &::-webkit-scrollbar {
    width: 6px;
  }

  /* Track */
  &::-webkit-scrollbar-track {
    background: rgba(255, 255, 255, 0.05);
    border-radius: 12px;
  }

  /* Handle */
  &::-webkit-scrollbar-thumb {
    background: rgba(255, 255, 255, 0.1);
    border-radius: 12px;
  }

  /* Handle on hover */
  &::-webkit-scrollbar-thumb:hover {
    background: rgba(255, 255, 255, 0.15);
  }
`;
