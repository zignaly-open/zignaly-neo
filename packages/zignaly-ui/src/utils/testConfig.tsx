import { render } from "@testing-library/react";
import { ThemeProvider as ThemeProviderMui } from "@mui/material";
import { ThemeProvider, ThemeProviderMui as ThemeInheritorMui } from "../index";
import React from "react";
import { getZignalyThemeExport } from "../theme";
import mediaQuery from "css-mediaquery";
import { JestI18nextDecorator } from "./dummyI18nextProvider";

export const renderWithProvidersUi = (ui: JSX.Element, renderOptions = {}) => {
  const { mui, legacyStyledComponentsDoNotUse } = getZignalyThemeExport();
  return render(ui, {
    wrapper: ({ children }: { children: JSX.Element }) => (
      <JestI18nextDecorator>
        <ThemeProvider theme={legacyStyledComponentsDoNotUse}>
          <ThemeInheritorMui theme={mui}>
            <ThemeProviderMui theme={mui}>{children}</ThemeProviderMui>
          </ThemeInheritorMui>
        </ThemeProvider>
      </JestI18nextDecorator>
    ),
    ...renderOptions,
  });
};

const createMatchMedia = (width: number) => {
  return (query: string) => {
    return {
      matches: mediaQuery.match(query, { width }),
      media: "",
      addListener: () => {},
      removeListener: () => {},
      onchange: () => {},
      addEventListener: () => {},
      removeEventListener: () => {},
      dispatchEvent: () => true,
    };
  };
};

export const resizeScreenSize = (width: number) => {
  window.matchMedia = createMatchMedia(width);
};
