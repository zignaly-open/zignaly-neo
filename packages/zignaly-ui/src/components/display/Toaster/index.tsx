// Dependencies
import React from "react";
import { ToasterProps } from "./types";
import * as styled from "./styles";
import Typography from "../Typography";
import { ReactComponent as SuccessIcon } from "assets/icons/check-icon.svg";
import { ReactComponent as ErrorAlertIcon } from "assets/icons/error-alert-icon.svg";
import { dark } from "theme";

const Toaster = ({ variant = "error", size = "large", caption = "" }: ToasterProps) => {
  return (
    <styled.ToastContainer variant={variant} size={size}>
      {variant === "error" ? (
        <styled.IconContainer>
          <ErrorAlertIcon color={dark.redGraphOrError} />
        </styled.IconContainer>
      ) : (
        <styled.IconContainer>
          <SuccessIcon color={dark.greenGraph} />
        </styled.IconContainer>
      )}
      <styled.Caption>
        <Typography variant={size === "large" ? "body1" : "body2"} weight="regular">
          {caption}
        </Typography>
      </styled.Caption>
    </styled.ToastContainer>
  );
};

export default Toaster;
