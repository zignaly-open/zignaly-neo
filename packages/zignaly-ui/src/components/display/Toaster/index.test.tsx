import React from "react";
import Toaster from "./index";
import { renderWithProvidersUi } from "../../../utils/testConfig";

describe("components/display/Toaster", () => {
  it("renders with default props", async () => {
    const { container } = renderWithProvidersUi(<Toaster id={"toaster"} />);

    const toaster = container.querySelector("#toaster") as Element;
    expect(toaster).toBeVisible();
    const toasterStyles = getComputedStyle(toaster);
    expect(toaster).toMatchSnapshot();
    expect(toasterStyles).toMatchSnapshot();
  });

  it("renders with small size prop", async () => {
    const { container } = renderWithProvidersUi(<Toaster size={"small"} id={"toaster"} />);

    const toaster = container.querySelector("#toaster") as Element;
    expect(toaster).toBeVisible();
    const toasterStyles = getComputedStyle(toaster);
    expect(toaster).toMatchSnapshot();
    expect(toasterStyles).toMatchSnapshot();
  });

  it("renders with success variant prop", async () => {
    const { container } = renderWithProvidersUi(<Toaster variant={"success"} id={"toaster"} />);

    const toaster = container.querySelector("#toaster") as Element;
    expect(toaster).toBeVisible();
    const toasterStyles = getComputedStyle(toaster);
    expect(toaster).toMatchSnapshot();
    expect(toasterStyles).toMatchSnapshot();
  });

  it("renders with info variant prop", async () => {
    const { container } = renderWithProvidersUi(<Toaster variant={"info"} id={"toaster"} />);

    const toaster = container.querySelector("#toaster") as Element;
    expect(toaster).toBeVisible();
    const toasterStyles = getComputedStyle(toaster);
    expect(toaster).toMatchSnapshot();
    expect(toasterStyles).toMatchSnapshot();
  });
});
