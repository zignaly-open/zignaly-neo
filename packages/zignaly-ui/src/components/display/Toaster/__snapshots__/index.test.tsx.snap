// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/display/Toaster renders with default props 1`] = `
<div
  class="sc-dkzDqf cPWBcW"
  id="toaster"
>
  <div
    class="sc-gsnTZi dwqIQN"
  >
    <svg
      color="#CC3993"
    />
  </div>
  <div
    class="sc-bczRLJ chEsjx"
  >
    <span
      class="MuiTypography-root MuiTypography-body1 css-pz04qi-MuiTypography-root"
    />
  </div>
</div>
`;

exports[`components/display/Toaster renders with default props 2`] = `
CSSStyleDeclaration {
  "0": "display",
  "1": "flex-direction",
  "10": "border",
  "11": "background",
  "12": "visibility",
  "2": "align-items",
  "3": "justify-content",
  "4": "box-sizing",
  "5": "box-shadow",
  "6": "border-radius",
  "7": "position",
  "8": "padding",
  "9": "height",
  "_importants": {
    "align-items": "",
    "background": "",
    "border": "",
    "border-radius": "",
    "box-shadow": "",
    "box-sizing": "",
    "display": "",
    "flex-direction": "",
    "height": "",
    "justify-content": "",
    "padding": "",
    "position": "",
    "visibility": undefined,
  },
  "_length": 13,
  "_onChange": [Function],
  "_values": {
    "align-items": "center",
    "background": "rgb(35, 22, 48)",
    "background-color": "rgb(35, 22, 48)",
    "border": "1px solid #231630",
    "border-bottom": "1px solid #231630",
    "border-bottom-color": "#231630",
    "border-bottom-style": "solid",
    "border-bottom-width": "1px",
    "border-color": "#231630",
    "border-left": "1px solid #231630",
    "border-left-color": "#231630",
    "border-left-style": "solid",
    "border-left-width": "1px",
    "border-radius": "5px",
    "border-right": "1px solid #231630",
    "border-right-color": "#231630",
    "border-right-style": "solid",
    "border-right-width": "1px",
    "border-style": "solid",
    "border-top": "1px solid #231630",
    "border-top-color": "#231630",
    "border-top-style": "solid",
    "border-top-width": "1px",
    "border-width": "1px",
    "box-shadow": "4px 6px 4px 44",
    "box-sizing": "border-box",
    "display": "flex",
    "flex-direction": "row",
    "height": "48px",
    "justify-content": "center",
    "padding": "11px 24px",
    "padding-bottom": "11px",
    "padding-left": "24px",
    "padding-right": "24px",
    "padding-top": "11px",
    "position": "absolute",
    "visibility": "visible",
  },
}
`;

exports[`components/display/Toaster renders with info variant prop 1`] = `
<div
  class="sc-dkzDqf dtZXct"
  id="toaster"
>
  <div
    class="sc-gsnTZi dwqIQN"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="InfoIcon"
      fill="#35334A"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
      />
    </svg>
  </div>
  <div
    class="sc-bczRLJ chEsjx"
  >
    <span
      class="MuiTypography-root MuiTypography-body1 css-fwwn99-MuiTypography-root"
    />
  </div>
</div>
`;

exports[`components/display/Toaster renders with info variant prop 2`] = `
CSSStyleDeclaration {
  "0": "display",
  "1": "flex-direction",
  "10": "border",
  "11": "background",
  "12": "visibility",
  "2": "align-items",
  "3": "justify-content",
  "4": "box-sizing",
  "5": "box-shadow",
  "6": "border-radius",
  "7": "position",
  "8": "padding",
  "9": "height",
  "_importants": {
    "align-items": "",
    "background": "",
    "border": "",
    "border-radius": "",
    "box-shadow": "",
    "box-sizing": "",
    "display": "",
    "flex-direction": "",
    "height": "",
    "justify-content": "",
    "padding": "",
    "position": "",
    "visibility": undefined,
  },
  "_length": 13,
  "_onChange": [Function],
  "_values": {
    "align-items": "center",
    "background": "rgb(53, 51, 74)",
    "background-color": "rgb(53, 51, 74)",
    "border": "1px solid #35334a",
    "border-bottom": "1px solid #35334a",
    "border-bottom-color": "#35334a",
    "border-bottom-style": "solid",
    "border-bottom-width": "1px",
    "border-color": "#35334a",
    "border-left": "1px solid #35334a",
    "border-left-color": "#35334a",
    "border-left-style": "solid",
    "border-left-width": "1px",
    "border-radius": "5px",
    "border-right": "1px solid #35334a",
    "border-right-color": "#35334a",
    "border-right-style": "solid",
    "border-right-width": "1px",
    "border-style": "solid",
    "border-top": "1px solid #35334a",
    "border-top-color": "#35334a",
    "border-top-style": "solid",
    "border-top-width": "1px",
    "border-width": "1px",
    "box-shadow": "4px 6px 4px 44",
    "box-sizing": "border-box",
    "display": "flex",
    "flex-direction": "row",
    "height": "48px",
    "justify-content": "center",
    "padding": "11px 24px",
    "padding-bottom": "11px",
    "padding-left": "24px",
    "padding-right": "24px",
    "padding-top": "11px",
    "position": "absolute",
    "visibility": "visible",
  },
}
`;

exports[`components/display/Toaster renders with small size prop 1`] = `
<div
  class="sc-dkzDqf hnTGNs"
  id="toaster"
>
  <div
    class="sc-gsnTZi dwqIQN"
  >
    <svg
      color="#CC3993"
    />
  </div>
  <div
    class="sc-bczRLJ chEsjx"
  >
    <span
      class="MuiTypography-root MuiTypography-body2 css-17qoo2p-MuiTypography-root"
    />
  </div>
</div>
`;

exports[`components/display/Toaster renders with small size prop 2`] = `
CSSStyleDeclaration {
  "0": "display",
  "1": "flex-direction",
  "10": "border",
  "11": "background",
  "12": "visibility",
  "2": "align-items",
  "3": "justify-content",
  "4": "box-sizing",
  "5": "box-shadow",
  "6": "border-radius",
  "7": "position",
  "8": "padding",
  "9": "height",
  "_importants": {
    "align-items": "",
    "background": "",
    "border": "",
    "border-radius": "",
    "box-shadow": "",
    "box-sizing": "",
    "display": "",
    "flex-direction": "",
    "height": "",
    "justify-content": "",
    "padding": "",
    "position": "",
    "visibility": undefined,
  },
  "_length": 13,
  "_onChange": [Function],
  "_values": {
    "align-items": "center",
    "background": "rgb(35, 22, 48)",
    "background-color": "rgb(35, 22, 48)",
    "border": "1px solid #231630",
    "border-bottom": "1px solid #231630",
    "border-bottom-color": "#231630",
    "border-bottom-style": "solid",
    "border-bottom-width": "1px",
    "border-color": "#231630",
    "border-left": "1px solid #231630",
    "border-left-color": "#231630",
    "border-left-style": "solid",
    "border-left-width": "1px",
    "border-radius": "5px",
    "border-right": "1px solid #231630",
    "border-right-color": "#231630",
    "border-right-style": "solid",
    "border-right-width": "1px",
    "border-style": "solid",
    "border-top": "1px solid #231630",
    "border-top-color": "#231630",
    "border-top-style": "solid",
    "border-top-width": "1px",
    "border-width": "1px",
    "box-shadow": "4px 6px 4px 44",
    "box-sizing": "border-box",
    "display": "flex",
    "flex-direction": "row",
    "height": "32px",
    "justify-content": "center",
    "padding": "4px 18px",
    "padding-bottom": "4px",
    "padding-left": "18px",
    "padding-right": "18px",
    "padding-top": "4px",
    "position": "absolute",
    "visibility": "visible",
  },
}
`;

exports[`components/display/Toaster renders with success variant prop 1`] = `
<div
  class="sc-dkzDqf vTvCu"
  id="toaster"
>
  <div
    class="sc-gsnTZi dwqIQN"
  >
    <svg
      color="#26c496"
    />
  </div>
  <div
    class="sc-bczRLJ chEsjx"
  >
    <span
      class="MuiTypography-root MuiTypography-body1 css-xnxsde-MuiTypography-root"
    />
  </div>
</div>
`;

exports[`components/display/Toaster renders with success variant prop 2`] = `
CSSStyleDeclaration {
  "0": "display",
  "1": "flex-direction",
  "10": "border",
  "11": "background",
  "12": "visibility",
  "2": "align-items",
  "3": "justify-content",
  "4": "box-sizing",
  "5": "box-shadow",
  "6": "border-radius",
  "7": "position",
  "8": "padding",
  "9": "height",
  "_importants": {
    "align-items": "",
    "background": "",
    "border": "",
    "border-radius": "",
    "box-shadow": "",
    "box-sizing": "",
    "display": "",
    "flex-direction": "",
    "height": "",
    "justify-content": "",
    "padding": "",
    "position": "",
    "visibility": undefined,
  },
  "_length": 13,
  "_onChange": [Function],
  "_values": {
    "align-items": "center",
    "background": "rgb(18, 36, 49)",
    "background-color": "rgb(18, 36, 49)",
    "border": "1px solid #122431",
    "border-bottom": "1px solid #122431",
    "border-bottom-color": "#122431",
    "border-bottom-style": "solid",
    "border-bottom-width": "1px",
    "border-color": "#122431",
    "border-left": "1px solid #122431",
    "border-left-color": "#122431",
    "border-left-style": "solid",
    "border-left-width": "1px",
    "border-radius": "5px",
    "border-right": "1px solid #122431",
    "border-right-color": "#122431",
    "border-right-style": "solid",
    "border-right-width": "1px",
    "border-style": "solid",
    "border-top": "1px solid #122431",
    "border-top-color": "#122431",
    "border-top-style": "solid",
    "border-top-width": "1px",
    "border-width": "1px",
    "box-shadow": "4px 6px 4px 44",
    "box-sizing": "border-box",
    "display": "flex",
    "flex-direction": "row",
    "height": "48px",
    "justify-content": "center",
    "padding": "11px 24px",
    "padding-bottom": "11px",
    "padding-left": "24px",
    "padding-right": "24px",
    "padding-top": "11px",
    "position": "absolute",
    "visibility": "visible",
  },
}
`;
